datalad/support/archives.py

```diff
@@ -314,9 +155,9 @@ def __init__(self, toppath=None, persistent=False):
                 self._made_path = True
                 os.makedirs(path)
                 lgr.debug("Cache initialized")
-            except:
+            except Exception as e:
                 lgr.error("Failed to initialize cached under %s" % path)
-                raise
+                raise e
```

Comment:
Since except is merely for logging, I would say that
```suggestion
                raise
```
would remain a preferred way to simply reraise exception, since otherwise it would use this line as the source of exception while also mentioning original one (in python3 only):
```
/tmp > cat /tmp/1.py
try:
    raise ValueError("raising on line 2")
except Exception as e:
    raise e

/tmp > python3 /tmp/1.py
Traceback (most recent call last):
  File "/tmp/1.py", line 4, in <module>
    raise e
  File "/tmp/1.py", line 2, in <module>
    raise ValueError("raising on line 2")
ValueError: raising on line 2
```
imho just adds confusion and probably would somewhat interfere with our logging of exceptions (would point to this re-raised line)

Reply:
Will do.

Reply:
Done